utils/python/CIME/XML/testlist.py

```diff
@@ -80,9 +86,14 @@ def _get_testsv2(self, machine=None, category=None, compiler=None):
                         else:
                             this_test[key] = value
 
+                    optionnodes = self.get_nodes("option", root=tnode)
```

Comment:
It looks to me like you're trying to say: Get all options attached to this specific machine/compiler (with the pre-existing code block below), and get all options that are at a higher level (with this new code block). However, this doesn't seem to work as intended: This new code block seems to pick up _all_ options, even those nested under a specific machine/compiler.

I tested this with this testlist:

```xml
<?xml version="1.0"?>
<testlist version="2.0">
  <test name="SMS" grid="ne30_g16" compset="X">
    <machines>
      <machine name="yellowstone" compiler="intel" category="bill">
      </machine>
      <machine name="yellowstone" compiler="pgi" category="bill">
        <options>
          <option name="wallclock">00:30:00</option>
        </options>
      </machine>
    </machines>
  </test>
</testlist>
```

And then:

```
./create_test --no-build --xml-category bill --test-root $scratch/tests_machopts_1004 --xml-testlist /glade/p/work/sacks/testing/testlist_machopts.xml
```

The result was that both the intel and pgi tests had their wallclock time set to 0:30.

I think that, for this new block of code, you need to somehow only find immediate children of tnode, rather than picking up any nodes rooted at tnode. Actually, that raises a question: Are these non-machine-specific options meant to be nested directly under the <test> node, so siblings of <machines>? Or are they meant to appear within the <machines> block? The first option would be most intuitive to me, and would agree with my suggestion of finding immediate children of tnode. But testlist_allactive uses the second.

However this is done: I'd suggest backing out this change here, and submitting a separate PR with it. That would keep this current PR clean and simple (and ready to go). My main rationale is that this change will potentially change behavior for many of our tests - because the options defined in testlist_allactive will begin to be applied, whereas they weren't before. So if this causes any problems, it would be good to have this change isolated to its own PR.